### DISCUSSION TITLE
s3 amazon

### DISCUSSION THREAD

**Discussion by Marcioghiraldelli** [AzuraCast/AzuraCast#3869] [Q&A]
I have doubts on how to configure the amazon s3, I searched and did not find a tutorial on.,

**Comment by ItsMitchh** [comment]
https://www.azuracast.com/extending/s3-configuration.html#amazon-simple-storage-service-s3

Here is our basic tutorial on setting up Amazon S3.

**Comment by Marcioghiraldelli** [comment]
thanks for showing.
I couldn't find the dropbox.
I think the dropbox price is much better.
Does anyone have any solution for dropbox?
THAT DOES NOT require a lot of space on the server.

**Comment by ItsMitchh** [comment]
Correct there is no dropbox in that page, I may consider adding it this week.

**Comment by Marcioghiraldelli** [comment]
It would be nice to be able to insert dropbox as well.
Because value and space are much more important.

**Comment by Catskill909** [comment]
I'm having trouble myself with s3 configuration.
The 2 fields I'm not sure how to fill out are
- Path/Suffix
- Endpoint
I see configuring Endpoints in s3 docs. Is this what I should pursue?
any help to get me on track appreciated

**Comment by Catskill909** [comment]
So I was finally able to work this out. There is a post that shows close to a working S3 mount configuration but the "Path/Suffix" field still used the default example in the help, so that was confusing. The "Endpoint" is just the s3 url to your bucket you want mounted but without the Bucket name in the URL
I had tried using a slash in the Path/Suffix and that was resulting in an error. It turns out that putting nothing there did the trick! The bucket I wanted mounted is in the "Root" of my S3

![Screen Shot 2021-01-30 at 4 16 12 PM](https://user-images.githubusercontent.com/36337311/106368431-608a3500-6317-11eb-8933-136337e1d9e6.png)

**Reply by robz0** [reply]
S3 storage on Digital Ocean

If anyone uses a Digital Ocean s3 bucket, there is a something to be aware of, when copying and pasting endpoint from the Space itself, seen as Orgin Endpoint. 
<img width="812" alt="image" src="https://github.com/AzuraCast/AzuraCast/assets/18615915/8dcf887a-34d6-483f-a7c8-79c24d4b23af">

When adding to AzuraCast,  If the endpoint is shown as this-->

`https://blahblah.ams3.digitaloceanspaces.com
`
this means the space/bucket is called blahblah

Then you would-
In AzuraCast Add s3 storage Location
Path can be:
 / 
(for the first/root folder or path to other folder on the drive)

Quota is optional but if added your media storage location will show current use/quota. You can still login successfuly with nothing in this field.

Access Key:  and Secret: is created in API page from the Cloud Objects home page.  

The Endpoint can be found on the Space droplet itself.
but when entering in Azuracast, here's the rub -

in Endpoint field, you have to take the bucket name off (if you copied the endpoint from DO space), because AzuraCast joins them back together. so the entry for 

Endpoint, based on above is just:
https://ams3.digitaloceanspaces.com

then in Bucket name put the:  
blahblah (or whatever the bucket/space name is)

Region is: 
ams3 (based on above example, the bit before the DO domain name)

and Version stays as: 
latest

When you save changes AzuraCast with add the blahblah back to the enpoint bit and get in. Otherwise it will fail by trying to connect to `https://blahblah.blahblah.am3.digitaloceanspaces.com` (if you copy the whole endpoint from the DO spaces panel into the Endpoint field)

Posting this as it took a little while for me to work that out, as I may not be obvious. 

This works for Digtal Ocean but It may be the case wtih AWS also or other providers but I have only tried them. That's how I connected AzuraCast to a DigitalOcean S3 Spaces.
